server/util/config.py

```diff
@@ -43,11 +43,13 @@ def __init__(self, absolute_filepath):
         self.file_path = absolute_filepath
         self.variables = _variables_from_file(self.file_path)  # a dict from VARIABLE_NAME to value
 
-    def get(self, key):
+    def get(self, key, default=None):
         variable_name = key.upper()
         try:
             return os.environ[variable_name]
         except KeyError:
+            if default:
+                return default
```

Comment:
This `get` is custom command used e.g [here](https://github.com/CodeForAfrica/web-tools/blob/main/server/auth.py#L164) so I was trying to make it be same as other `get` commands i.e same way you can specify a default.

Reply:
I got that @esirK. What I meant was looking at the code, if the key isn't in the `env`, `default` will be returned before a check in `self.variables`. What I'm asking is, shouldn't it check in `env`, then `self.variables` and then and only then return `default` if not found.

Reply:
@kilemensi  I think both approaches can work but what I was thinking is if you call `get` and add a default to it then that's what you expect to be used incase that environment variable hasn't been set. If you don't know the default you want used then you don't provide a default and whatever is in the  `variables` gets used.

Reply:
Not sure I follow you @esirK. My uderstanding is `default` value in `get` means **if no value is set** for this variable, _then_ use this value. `config` can set values using `environment variables` or `file`.

The way the code is written now is if no value is set in _environment variable_ then use this value **even if** that same variable is set using _file_.

Reply:
I get you @kilemensi  but what I'm saying is assume 
1. `self.variables` which is created from a file e.g `app.defaults.config` has `URL` set to `http://example.com` and then
2. you do something like `get("URL", "http://12.0.0.1")`
Which `URL` would you expect to be used?

Reply:
http://example.com @esirK . That's what is **set**.

Reply:
If in `app.defaults.config` the `URL` **is** empty, then `http://12.0.0.1`

Reply:
Okay. Although in this case, I would assume you intend to override the default in `app.defaults.config` otherwise you could just call `get` without a new default.